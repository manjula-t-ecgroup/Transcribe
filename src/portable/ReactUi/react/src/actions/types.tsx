export const FETCH_USERS = 'FETCH_USERS';
export const SELECT_USER = 'SELECT_USER';