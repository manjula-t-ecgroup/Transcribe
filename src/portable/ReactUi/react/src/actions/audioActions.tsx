import Axios from 'axios';
import { COPY_AUDIO, INITIAL_TRANSCRIPTION, JUMP_CHANGE, PLAY_STATUS, PLAYSPEEDRATE_CHANGE,
    REPORT_POSITION, REQUEST_POSITION, SAVE_STATUS, SAVE_TOTAL_SECONDS, SUBMIT_STATUS } from './types';


export function playStatus(playing: boolean): any{
    return {
        payload: playing,
        type: PLAY_STATUS
    }
}

export function playSpeedRateChange(playSpeedRate: number): any{
    return {
        payload: playSpeedRate,
        type: PLAYSPEEDRATE_CHANGE
    }
}

export function jumpChange(jump: number): any{
    return {
        payload: jump,
        type: JUMP_CHANGE
    }
}

export function saveTotalSeconds(seconds: number): any {
    return {
        payload: seconds,
        type: SAVE_TOTAL_SECONDS
    }
}

export const reportPosition = (taskid: string, seconds: number) => (dispatch: any) => {
    dispatch({payload: seconds, type: REPORT_POSITION});
    Axios.put('/api/ReportPosition?task=' + taskid + '&position=' + seconds.toString());
}

export function requestPosition(): any {
    return {
        type: REQUEST_POSITION
    }
}

export function setInitialTranscription(status: boolean): any {
    return {
        payload: status,
        type: INITIAL_TRANSCRIPTION
    }
}

export function setSubmitted(submit: boolean): any{
    return {
        payload: submit,
        type: SUBMIT_STATUS
    }
}

export function setSaved(saved: boolean): any{
    return {
        payload: saved,
        type: SAVE_STATUS
    }
}

export const copyAudio = (task: string, project: string, audioFile: string, data: object) => (dispatch: any) => {
    dispatch({type: COPY_AUDIO});
    Axios.put('/api/CopyAudio?task=' + task + '&project=' + project+ '&audioFile=' + audioFile, data)
}