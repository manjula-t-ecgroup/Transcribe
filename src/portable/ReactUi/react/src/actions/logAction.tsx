import Axios from 'axios';

export const log = (message: string) => {
    Axios.put('/api/Log?' + message);
}
