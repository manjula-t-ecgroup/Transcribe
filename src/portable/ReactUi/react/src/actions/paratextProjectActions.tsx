import Axios from 'axios';
import { FETCH_PARATEXT_PROJECTS, SELECT_PARATEXT_PROJECT } from './types';

export const fetchParatextProjects = () => (dispatch: any) => {
    // tslint:disable-next-line:no-console
    console.log("in get")
    Axios.get('/api/GetParatextProjects').
        then(paratextProjects => {
            dispatch({
                payload: paratextProjects,
                type: FETCH_PARATEXT_PROJECTS
            });
        });
}

export function selectParatextProject(id: string): any{
    return {
        payload: id,
        type: SELECT_PARATEXT_PROJECT
    }
}