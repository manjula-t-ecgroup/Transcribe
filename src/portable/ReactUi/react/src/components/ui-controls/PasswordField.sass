@import "../../_variables.sass"

.PasswordField
    width: 448px
    height: 56px !important
    display: flex
    flex-direction: column
    background: transparent
    justify-content: flex-start
    padding-bottom: 24px
    padding-left: 35px
    text-align: left

    .bodyRow
        display: flex
        flex-direction: row
        background-color: rgba(255, 255, 255, 0.15)
        padding: 9px 12px 0 12px
        min-height: min-content
        justify-content: space-between

    .dataColumn
        display: flex
        flex-direction: column
        flex: 1 0 auto

    .clearButton
        display: flex
        flex: 0 0 auto

    .Caption
        font-family: $sans
        font-size: 12px
        height: 16px
        letter-spacing: 0.4px
        line-height: 16px

    input
        flex: 1 0 auto
        background: rgb(42, 82, 90)
        border: none
        font-family: $sans
        font-size: 16px
        height: 28px
        letter-spacing: 0.44px
    .DataColor
        color: $base2
    .CaptionGrey
        color: $base1

    .Message
        font-family: $sans
        font-size: 12px
        height: 16px
        letter-spacing: 0.4px
        line-height: 16px
        padding: 0 12px

    .BorderTopGreen
        border-top: 2px solid $greenLight

    .BorderTopRed
        border-top: 2px solid $red

    .CaptionGreen
        color: $greenLight

    .CaptionRed
        color: $red

    .Hide
        display: none
        
    

    .tickButton
        padding: 10px
        .OnHover:hover 
            -moz-border-radius: 5px
            border-radius: 5px
            -moz-box-shadow: 0px 0px 30px 0px $white
            box-shadow: 0px 0px 30px 0px $white
