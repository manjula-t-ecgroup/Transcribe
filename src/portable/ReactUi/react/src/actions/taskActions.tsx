import Axios from 'axios';
import { setSubmitted } from './audioActions';
import { ASSIGN_TASK_PENDING, COMPLETE_REVIEW_PENDING, COMPLETE_TRANSCRIPTION_PENDING, 
    FETCH_TASKS, FETCH_TRANSCRIPTION, SELECT_POPUP_TASK, SELECT_PROJECT, SELECT_TASK,  UNASSIGN_TASK_PENDING,
    UPDATE_TASK, WRITE_FULFILLED, WRITE_PENDING } from './types';
import { saveUserSetting } from './userActions';

export const assignTask = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: ASSIGN_TASK_PENDING});
    Axios.put('/api/TaskEvent?action=Assigned&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
        .then(dispatch(setSubmitted(false)))
}

export const unAssignTask = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: UNASSIGN_TASK_PENDING});
    Axios.put('/api/TaskEvent?action=Unassigned&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
}

export const completeTranscription = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: COMPLETE_TRANSCRIPTION_PENDING});
    Axios.put('/api/TaskEvent?action=TranscribeEnd&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
}

export const completeReview = (taskid: string, userid: string) => (dispatch: any) => {
    dispatch({type: COMPLETE_REVIEW_PENDING});
    Axios.put('/api/TaskEvent?action=ReviewEnd&task=' + taskid + '&user=' + userid)
        .then(dispatch(uploadTranscription(taskid, userid)))
}

export const uploadTranscription = (taskid: string, userid: string) => (dispatch: any) => {
    Axios.put('/api/TaskEvent?action=Upload&task=' + taskid + '&user=' + userid)
        .then(dispatch(fetchTasks(userid)))
}

export const writeTranscription = (taskid: string, length: number, lang: string, dir: string, data: any) => (dispatch: any) => {
    dispatch({type: WRITE_PENDING});
    Axios.put('/api/WriteTranscription?task=' + taskid + "&length=" + length.toString() + "&lang=" + lang + "&dir=" + dir, data)
        .then(dispatch({type: WRITE_FULFILLED}))
}

export const fetchTasks = (username: string) => (dispatch: any) => {
    Axios.get('/api/GetTasks?user=' + username)
        .then(tasks => {
            dispatch({
                payload: tasks,
                type: FETCH_TASKS
            });
        })
}

export const fetchTasksOfProject = (projectname: string) => (dispatch: any) => {
    Axios.get('/api/GetTasks?project=' + projectname)
        .then(tasks => {
            dispatch({
                payload: tasks,
                type: FETCH_TASKS
            });
            dispatch({
                payload: projectname,
                type: SELECT_PROJECT
            })
        })
    }

export function selectProject(id: string): any{
    return {
        payload: id,
        type: SELECT_PROJECT
    }
}

export const  selectTask = (user: string, id: string) => (dispatch:any) => {
    dispatch({
        payload: id,
        type: SELECT_TASK
    })
    dispatch(fetchTranscription(id));
    dispatch(saveUserSetting(user, "lastTask", id))
    dispatch(setSubmitted(false))
}

export const  selectPopupTask = (id: string) => (dispatch:any) => {
    dispatch({
        payload: id,
        type: SELECT_POPUP_TASK
    })
}

export const fetchTranscription = (taskid: string) => (dispatch: any) => {
    const part = taskid.split('.');
    Axios.get('/api/audio/' + part[0] + '.transcription').
    then(transcription => {
        dispatch({
            payload: transcription,
            type: FETCH_TRANSCRIPTION
        });
    });
}

export const updateTask = (task: string, project: string, query: string, data: object) => (dispatch: any) => {
    dispatch({type: UPDATE_TASK});
    Axios.put('/api/UpdateTask?task=' + task + '&project=' + project + query, data)
        .then(dispatch(fetchTasksOfProject(project)))
}
