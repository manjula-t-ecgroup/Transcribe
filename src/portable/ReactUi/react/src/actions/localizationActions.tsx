import Axios from 'axios';
import { log } from '../actions/logAction';
import { FETCH_LOCALIZATION } from './types';


export const fetchLocalization = () => (dispatch: any) => {
    Axios.get('/localization/strings.json')
        .then(strings => {
            dispatch({
                payload: strings,
                type: FETCH_LOCALIZATION
            });
        })
        .catch((reason: any) => {
            dispatch(log(JSON.stringify(reason) + " fetch localization"))
        });
}
