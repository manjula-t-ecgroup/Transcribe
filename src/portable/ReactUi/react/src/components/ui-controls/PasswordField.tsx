import * as React from 'react';
import './PasswordField.sass';

interface IProps {
    id?: string,
    caption: string,
    inputValue?: string,
    message?: string,
    show?: boolean,
    isReadOnly?: boolean,
    onChange?: (value: string) => any;
    setAdminPassword?: (password: string) => any;
}

export class PasswordField extends React.Component<IProps, any> {
    public state = {
        current: this.props.inputValue,
    }

    constructor(props: IProps) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
        this.handleClear = this.handleClear.bind(this);
        this.handleTick = this.handleTick.bind(this);
    }

    public handleTick(event: any) {
        // tslint:disable-next-line:no-console
        console.log("Tick is clicked!!");
    }

    public handleChange(event: any) {
        this.setState({current: event.target.value });
        if (this.props.setAdminPassword !== undefined){
            this.props.setAdminPassword(event.target.value);
        }
    }

    public handleClear(event: any) {
        if (!this.props.isReadOnly) {
            this.setState({current: "" })
            if (this.props.setAdminPassword !== undefined){
                this.props.setAdminPassword("");
            }
        }
    }

    public render() {
        const { current } = this.state;
        const { caption, id, isReadOnly, message, show } = this.props;
        const errorMessage = current !== "" && message? message: "";
        const messageStyle = (errorMessage.length > 0)? "Message CaptionRed BorderTopRed": "Message BorderTopGreen";
        const captionStyle = (errorMessage.length > 0)? "Caption CaptionRed": "Caption CaptionGreen";
        const clearStyle = isReadOnly || current === ""? "Hide": "clearButton";
        const tickStyle = show ? "tickButton": "tickButton Hide";
        return (
            <div id={id} className="PasswordField">
                <div className="bodyRow">
                    <div className="dataColumn">
                        <label  className={captionStyle}>{current && current !== ""? caption: ""}</label>
                        <input
                            readOnly={isReadOnly}
                            className="DataColor"
                            type="password"
                            maxLength={12}
                            minLength={6}
                            value={current}
                            placeholder={caption}
                            onChange={this.handleChange}/>
                    </div>
                    <div className={tickStyle} onClick={this.handleTick}>
                        <img src="/assets/Tick.svg" alt="E" />
                    </div>
                    <div className={clearStyle} onClick={this.handleClear}>
                        <img src="/assets/Clear.svg" alt="X" />
                    </div>
                </div>
                <div className={messageStyle}>
                    <label>{errorMessage}</label>
                </div>
            </div>
        )
    }
}

export default (PasswordField);