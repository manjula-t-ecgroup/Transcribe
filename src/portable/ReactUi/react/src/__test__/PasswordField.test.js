/* jshint esversion: 6 */
import * as React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import * as actions from '../actions/audioActions';
import PasswordField from '../components/ui-controls/PasswordField';

configure({ adapter: new Adapter() })

// Snapshot for PasswordField
describe('>>>Control: PasswordField --- Snapshot',()=>{
    const minProps = {
        caption: "Password"
    };

    it('+++capturing Snapshot of PasswordField', () => {
        const renderedValue =  renderer.create(<PasswordField {...minProps}/>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: PasswordField', () => {
    let wrapper;
    const minProps = {
        caption: "Password"
    };

    beforeEach(()=>{
        wrapper = shallow(<PasswordField {...minProps} />)
    })

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });
});
