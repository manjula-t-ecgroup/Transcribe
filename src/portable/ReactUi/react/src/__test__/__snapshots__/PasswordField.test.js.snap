// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: PasswordField --- Snapshot +++capturing Snapshot of PasswordField 1`] = `
<div
  className="PasswordField"
  id={undefined}
>
  <div
    className="bodyRow"
  >
    <div
      className="dataColumn"
    >
      <label
        className="Caption CaptionGreen"
      >
        
      </label>
      <input
        className="DataColor"
        maxLength={12}
        minLength={6}
        onChange={[Function]}
        placeholder="Password"
        readOnly={undefined}
        type="password"
        value={undefined}
      />
    </div>
    <div
      className="tickButton Hide"
      onClick={[Function]}
    >
      <img
        alt="E"
        src="/assets/Tick.svg"
      />
    </div>
    <div
      className="clearButton"
      onClick={[Function]}
    >
      <img
        alt="X"
        src="/assets/Clear.svg"
      />
    </div>
  </div>
  <div
    className="Message BorderTopGreen"
  >
    <label>
      
    </label>
  </div>
</div>
`;
